File: TLBB1.5/tabletools/CodeTable/Java/xjgame/table/Table_Language.java
//This code create by CodeEngine mrd.cyou.com ,don't modify
package xjgame.table;
import java.util.ArrayList;
 import java.util.HashMap;

public class Table_Language implements ITableOperate {
 public final static String TAB_FILE_DATA = "/language.txt";
 public static final int ID_CHINESE = 0 ;
public static final int ID_ENGLISH = 1 ;
public static final int MAX_RECORD = 2 ;

 public String GetInstanceFile(){return TAB_FILE_DATA; }

private String m_Chinese;
 public String GetChinese() { return m_Chinese;}

private String m_English;
 public String GetEnglish() { return m_English;}

public boolean LoadTable(HashMap<Integer, Table_Language> _tab) throws TableException
 {
 if(!TableManager.ReaderPList(GetInstanceFile(),this,_tab))
 {
 throw TableException.ErrorReader("Load File{0} Fail!!!",GetInstanceFile());
 }
 return true;
 }

 @Override
 public void SerializableTable(ArrayList<String> valuesList,String skey,Object obj) throws TableException {
 @SuppressWarnings("unchecked")
 HashMap<Integer, Table_Language> _hash = (HashMap<Integer, Table_Language> )obj;
 if (skey.isEmpty())
 {
 throw TableException.ErrorReader("Read File"+GetInstanceFile()+" as key is Empty Fail!!!");
 }

 if (MAX_RECORD!=valuesList.size())
 {
 throw TableException.ErrorReader("Load "+GetInstanceFile()+" error as CodeSize:"+MAX_RECORD+" not Equal DataSize:"+valuesList.size());
 }
 try { 
 Integer nKey = Integer.parseInt(skey);
 Table_Language _values = new Table_Language();
 _values.m_Chinese =  (String)valuesList.get(ID_CHINESE);
_values.m_English =  (String)valuesList.get(ID_ENGLISH);

 _hash.put(nKey,_values);
 } catch (NumberFormatException e) {
 throw TableException.ErrorReader("Load "+GetInstanceFile()+" error as :" + e.getMessage());
 }
 }


}

File: TLBB1.5/Client/Assets/Client/Scripts/Table/Table_Language.cs
//This code create by CodeEngine mrd.cyou.com ,don't modify
using System;
 using System.Collections.Generic;
 using System.Collections;

namespace GCGame.Table{

[Serializable]
 public class Tab_Language : ITableOperate
{ private const string TAB_FILE_DATA = "language.txt";
 public enum _ID
 {
 INVLAID_INDEX=-1,
ID_CHINESE,
ID_ENGLISH,
MAX_RECORD
 }
 public string GetInstanceFile(){return TAB_FILE_DATA; }

private string m_Chinese;
 public string Chinese { get{ return m_Chinese;}}

private string m_English;
 public string English { get{ return m_English;}}

public bool LoadTable(Hashtable _tab)
 {
 if(!TableManager.ReaderPList(GetInstanceFile(),SerializableTable,_tab))
 {
 throw TableException.ErrorReader("Load File{0} Fail!!!",GetInstanceFile());
 }
 return true;
 }
 public void SerializableTable(ArrayList valuesList,string skey,Hashtable _hash)
 {
 if (string.IsNullOrEmpty(skey))
 {
 throw TableException.ErrorReader("Read File{0} as key is Empty Fail!!!", GetInstanceFile());
 }

 if ((int)_ID.MAX_RECORD!=valuesList.Count)
 {
 throw TableException.ErrorReader("Load {0} error as CodeSize:{1} not Equal DataSize:{2}", GetInstanceFile(),_ID.MAX_RECORD,valuesList.Count);
 }
 Int32 nKey = Convert.ToInt32(skey);
 Tab_Language _values = new Tab_Language();
 _values.m_Chinese =  valuesList[(int)_ID.ID_CHINESE] as string;
_values.m_English =  valuesList[(int)_ID.ID_ENGLISH] as string;

 _hash.Add(nKey,_values); }


}
}

